ui/messages/html/parser.go

```diff
@@ -170,16 +175,33 @@ func (parser *htmlParser) blockquoteToEntity(node *html.Node) Entity {
 }
 
 func (parser *htmlParser) linkToEntity(node *html.Node) Entity {
+	href := parser.getAttribute(node, "href")
+
 	entity := &ContainerEntity{
 		BaseEntity: &BaseEntity{
 			Tag: "a",
 		},
 		Children: parser.nodeToEntities(node.FirstChild),
 	}
-	href := parser.getAttribute(node, "href")
+
 	if len(href) == 0 {
 		return entity
 	}
+
+	if len(entity.Children) == 1 {
+		entity, ok := entity.Children[0].(*TextEntity)
+		if ok && entity.Text == href {
+			parser.sameURL = true
+		}
+	}
+
+	if !parser.prefs.DisableShowurls && !parser.mautrixnolink && !parser.sameURL {
+		entity.Children = append(
+			[]Entity{NewTextEntity("(" + href + ") ")},
+			parser.nodeToEntities(node.FirstChild)...,
+		)
```

Comment:
This should probably be the other way round.

```suggestion
		entity.Children = append(entity.Children, NewTextEntity(fmt.Sprintf(" (%s)", href)))
```